design/open/kv-query-tool.md

```diff
@@ -0,0 +1,45 @@
+# KV Store Querying Tool
+
+## Scope
+
+Our newly introduced KV Store stores the data in the form of [`Partition Key`, `Key`, `Value`] triplets. The `partition key` and `key` fields are strings stored as byte arrays and the `value` is an object (various types) serialized as protobuf and stored as byte array.
+This data is not easily interpretable for human viewer, whereas our former SQL DB stored the data (mostly) as human readable data, making it easier for developers to look into the DB, and debug various situations. It also provided the same ease to insert data or alter existing data in the DB for experimentation and, in some cases, as a fix
+With our new KV Store this became less trivial and this proposal comes to offer a solution for that
+
+## Introducing `lakefs kv`
+
+The `lakefs` server contains all the logic needed to correctly read and update the KV Store. It generates objects into byte arrays with their protobuf encoding, on one hand, and rebuilds the objects from byte arrays with protobuf on the other, and, it holds all the logic to generate the relevant keys (e.g. `partition key` from a repository object, branch `key` from the branch object etc.)
+We will elevate this logic and into a new command that will provide the ability to read data items, scan keys and update values, using a new `lakefs` command - `kv`
+```
+lakefs kv <cmd> <args>
+```
+
+### Get/Scan/Set
```

Comment:
In my opinion supporting Set and Delete are risky and not necessary. In the very rare occasion where an entry needs to be modified it should be done directly on the DB and by a lakeFS engineer.
Any situation that requires modifying multiple key (via scan and set) for example, is irrelevant since it will probably already be a catastrophe

Reply:
I tend to agree that P0 would be Scan/Get (or even just Scan) and the rest can be added when needed.

Reply:
I'm OK with Scan/Get as a first phase.
I do think that altering data will be proven useful, at least for testing and development attempts.
We can also "protect" it by requiring an `Are you sure?` repsponse, etc. - WDYT?

Will update the doc to reflect the prioritization changes

Reply:
Definitely requires a prompt, also I'm for making it an internal dev tool.

Reply:
Agree. The entire tool is aimed to be internal

Reply:
Added to the document